stat/spatial/spatial.go

```diff
@@ -43,11 +42,21 @@ func GetisOrdGStar(i int, data, weights []float64, locality mat.Matrix) float64
 	n := float64(len(data))
 	mean, std := stat.MeanStdDev(data, weights)
 	var dwd, dww, sw float64
-	for j, v := range data {
-		w := locality.At(i, j)
-		sw += w
-		dwd += w * v
-		dww += w * w
+	if band, ok := locality.(mat.Banded); ok {
```

Comment:
Are you sure? It seems reasonable that one would want to use this on some kind of taurus topology, where the matrix would be banded, except with some data on the corners. This would be really easy to implement with a banded matrix plus some extra storage, but would not be compatible with this type switch.